### ISSUE TITLE
Deprecated Features Used Warning

### ISSUE THREAD

**Issue by kylegoch** [chef-cookbooks/aws#270]
### Cookbook version
4.2.0

### Chef-client version
12.18.51

In Test Kitchen, I am getting a `Deprecated Features Used` warning:

```
Deprecated features used!
  Property `sensitive` of resource `aws_s3_file` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/aws/resources/s3_file.rb:34:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `retries` of resource `aws_s3_file` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/aws/resources/s3_file.rb:35:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `retry_delay` of resource `aws_s3_file` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/aws/resources/s3_file.rb:36:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `sensitive` of resource `apt_repository` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/apt/resources/repository.rb:54:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `supports` of resource `chef_handler` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/chef_handler/resources/default.rb:31:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `sensitive` of resource `yum_repository` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/yum/resources/repository.rb:59:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
```

Might already be aware, but passing along none the less.

**Comment by tas50** [comment]
@kylegoch Can you try this on Chef 12.19. We think we eliminated those deprecations for this scenario, but take a look.

**Comment by kylegoch** [comment]
Just tried in test-kitchen and got it again:
```
Deprecated features used!
  Property `sensitive` of resource `aws_s3_file` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/aws/resources/s3_file.rb:34:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `retries` of resource `aws_s3_file` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/aws/resources/s3_file.rb:35:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
  Property `retry_delay` of resource `aws_s3_file` overwrites an existing method. Please use a different property name. This will raise an exception in Chef 13. at 1 location:
    - /opt/kitchen/cache/cookbooks/aws/resources/s3_file.rb:36:in `class_from_file'
   See https://docs.chef.io/deprecations_property_name_collision.html for further details.
```

Chef: 12.19.35
Cookbook: 4.2.0

**Comment by tas50** [comment]
I'll see what I can do there then. Thanks for the info

**Comment by tas50** [comment]
Fixed in 4.2.1. Thanks for the report

**Comment by kylegoch** [comment]
Yep. It's gone. Thanks!!!
